TF_Tooling_Web/src/Connection/TouchFreeServiceTypes.ts

```diff
@@ -71,6 +73,15 @@ export enum HandPresenceState {
     PROCESSED,
 }
 
+// Enum: InteractionZoneState
+// HAND_ENTERED - Sent when the "active" hand enters the interaction zone
+// HAND_EXITED - Sent when the "active" hand leaves the interaction zone
+export enum InteractionZoneState {
```

Comment:
🤷 Consider using two separate union types for this instead of an enum. I know it's a copy of another implementation but I don't think it's a great way to do things.
i.e.
```ts
export type InteractionZoneState = "HAND_ENTERED" | "HAND_EXITED";
export type InteractionZoneEventStatus = "PROCESSED" | "UNPROCESSED";
```
The second one could be a general purpose type instead of specific to interaction zone for other usages of this pattern.

Reply:
Only issue with this is that the service uses an enum for the state so it comes through as a number. The tooling would then have to do a translation which feels like it'd just be a bit of bloat. However, I do like the type implementation.

Instead, we could do this:
```ts
export type EventStatus = "PROCESSED" | "UNPROCESSED";
export enum InteractionZoneState {
    HAND_ENTERED,
    HAND_EXITED,
}
```
What do you think of that as a halfway house?

Reply:
I've pushed an update to this which _kind of_ implements what I've said above. Would love to get your thoughts!

Reply:
Good compromise - personally I'd go for a mapped type with the string literal union to translate but this implementation addresses the main issue where state was being overloaded.